rgrow/src/models/sdc1d.rs

```diff
@@ -0,0 +1,805 @@
+        self.generate_friends();
+    }
+
+    fn generate_friends(&mut self) {
+        let mut friends_btm = HashMap::new();
+        for (t, &b) in self
+            .glues
+            .index_axis(ndarray::Axis(1), BOTTOM_GLUE_INDEX)
+            .indexed_iter()
+        {
+            friends_btm
+                .entry(b)
+                .or_insert(HashSet::new())
+                .insert(t as u32);
+        }
+        self.friends_btm = friends_btm;
+    }
+
+    /// The strenght of glues a, b is given by:
+    ///
+    /// G(a, b) =  G_(37) (a,b) - (T - 37) * S(a, b)
+    fn generate_glue_matrix(&mut self) {
+        self.glue_links = &self.delta_g_matrix - (self.temperature - 37.0) * &self.entropy_matrix;
+    }
+
+    pub fn change_temperature_to(&mut self, kelvin: f64) {
+        self.temperature = kelvin;
+        self.update_system();
+    }
+
+    fn polymer_update<S: State + ?Sized>(&self, points: &Vec<PointSafe2>, state: &mut S) {
+        let mut points_to_update = points
+            .iter()
+            .flat_map(|&point| {
+                [
+                    PointSafeHere(point.0),
+                    state.move_sa_w(point),
+                    state.move_sa_e(point),
+                ]
+            })
+            .collect::<Vec<PointSafeHere>>();
+
+        points_to_update.sort_unstable();
+        points_to_update.dedup();
+        self.update_points(state, &points_to_update)
+    }
+
+    fn update_monomer_point<S: State + ?Sized>(&self, state: &mut S, scaffold_point: &PointSafe2) {
+        let points = [
+            state.move_sa_w(*scaffold_point),
+            state.move_sa_e(*scaffold_point),
+            PointSafeHere(scaffold_point.0),
+        ]
+        .map(|point| (point, self.event_rate_at_point(state, point)));
+
+        state.update_multiple(&points);
+    }
+
+    /// Fill the energy_bonds array
+    fn fill_energy_array(&mut self) {
+        let num_of_strands = self.strand_names.len();
+
+        // For each *possible* pair of strands, calculate the energy bond
+        for strand_f in 0..(num_of_strands as usize) {
+            let (f_west_glue, f_btm_glue, f_east_glue) = {
+                let glues = self.glues.row(strand_f);
+                (
+                    glues[WEST_GLUE_INDEX],
+                    glues[BOTTOM_GLUE_INDEX],
+                    glues[EAST_GLUE_INDEX],
+                )
+            };
+
+            for strand_s in 0..(num_of_strands as usize) {
+                let (s_west_glue, s_east_glue) = {
+                    let glues = self.glues.row(strand_s);
+                    (glues[WEST_GLUE_INDEX], glues[EAST_GLUE_INDEX])
+                };
+
+                // Calculate the energy between the two strands
+
+                // Case 1: First strands is to the west of second
+                // strand_f    strand_s
+                self.strand_energy_bonds[(strand_f, strand_s)] =
+                    -self.glue_links[(f_east_glue, s_west_glue)];
```

Comment:
If we're doing negative-is-stronger throughout, this, line 231 and line 236 should not have have negatives.